### PR TITLE
Resolve to absolute paths before checking for locality / use absolutes for caching / use cloud-files

### PR DESCRIPTION
**Pull request by dodamih**
Also updated `.gitignore` to add other Vim swap files.

### CODE REVIEW

**Comment by dodamih** [comment]
Finally ready.

File: zetta_utils/common/path.py [hunk 1]
```diff
@@ -14,3 +14,7 @@ def abspath(path: str) -> str:  # pragma: no cover
     if prefixes == ["file"]:
```
Comment (supersergiy) [inline zetta_utils/common/path.py:14]: `abspath` is getting a lot of logic now and should be tested

File: zetta_utils/geometry/bbox_strider.py [hunk 2]
```diff
@@ -166,9 +166,7 @@ def _attrs_post_init_nonexact(self) -> None:
             )
         )
         bbox_size_in_unit = self.bbox.end - self.bbox.start
-        # TODO: Type this variable again once typeguard bug has been fixed:
-        # https://github.com/agronholm/typeguard/issues/380
-        step_limits_raw = Vec3D[float](
+        step_limits_raw: Vec3D[float] = Vec3D[float](
```
Comment (supersergiy) [inline zetta_utils/geometry/bbox_strider.py:169]: Why does it need the explicit type annotation here?

File: zetta_utils/layer/volumetric/precomputed/precomputed.py [hunk 3]
```diff
@@ -1,49 +1,58 @@
...
 from typing import Any, Dict, Literal, Optional, Sequence, Tuple
 
 import attrs
 import cachetools
-import fsspec
-import fsspec.asyn
 from cachetools.keys import hashkey
 
-from zetta_utils.common import abspath
+# import fsspec
```
Comment (supersergiy) [inline zetta_utils/layer/volumetric/precomputed/precomputed.py:12]: Clean up?

**Review by supersergiy — APPROVED** [review]

**Pull request merged**